src/notebook/notebook/toolbar.ts

```diff
@@ -77,13 +77,13 @@ class NotebookToolbar extends Widget {
     if (index === -1) {
       layout.addChild(widget);
     } else {
-      layout.insertChild(index, widget);
+      layout.insertChild(index + 1, widget);
```

Comment:
Thanks. I thought I saw this bug when doing the toolbar demo, but hadn't investigated it yet...